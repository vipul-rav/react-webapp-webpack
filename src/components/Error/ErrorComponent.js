import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ErrorComponent extends Component {
  constructor(props) {
    super(props);
    this.state = { hasError: props.hasError || false };
  }

  componentDidCatch(error, info) {
    // Display fallback UI
    this.setState({ hasError: true });
    // eslint-disable-next-line no-console
    console.log(error);
    // eslint-disable-next-line no-console
    console.log(info);
  }
  render() {
    const { content } = this.props;
    if (this.state.hasError) {
      return (
        <div>
          <h1>{content.errorHeader}</h1>
          <div>
            <p> {content.errorText} </p>
          </div>
          <div>
            <button>{content.ExitText}</button>
          </div>
        </div>
      );
    }
    return this.props.children;
  }
}

ErrorComponent.propTypes = {
  content: PropTypes.object.isRequired,
  children: PropTypes.any,
  hasError: PropTypes.bool
};

export { ErrorComponent };
