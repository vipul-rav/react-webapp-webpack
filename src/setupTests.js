import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

let store;
beforeEach(() => {
  const mockStore = configureMockStore();
  store = mockStore();
});

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
