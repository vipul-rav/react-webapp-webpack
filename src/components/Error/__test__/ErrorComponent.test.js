import React from 'react';
import { render } from '../../../setupTests';
import { ErrorComponent } from '../ErrorComponent';
import configureMockStore from 'redux-mock-store';

const content = {
  ExitText: 'Exit',
  btnContinue: 'Continue',
  homePageText: 'Home Page'
};

const setup = (overridesProps) => {
  const props = {
    content: content,

    ...overridesProps
  };

  const mockStore = configureMockStore();
  const store = mockStore({
    config: {
      externalContent: content
    }
  });

  const { container, getByText } = render(
    <ErrorComponent {...props} store={store}>
      <div>test</div>
    </ErrorComponent>
  );

  return {
    props,
    container,
    getByText
  };
};

describe('Error Component', () => {
  it('should render - hasError=true', () => {
    const overrideProps = {
      hasError: true,
      content: {
        errorHeader: 'Technical Error',
        errorText: 'There are some technical error',
        ExitText: 'Exit'
      }
    };
    const { container } = setup(overrideProps);
    expect(container).toMatchSnapshot();
  });
  it('should render - hasError=false', () => {
    const overrideProps = {
      hasError: false,
      content: {
        errorHeader: 'Technical Error',
        errorText: 'There are some technical error',
        ExitText: 'Exit'
      }
    };
    const { container } = setup(overrideProps);
    expect(container).toMatchSnapshot();
  });
});
