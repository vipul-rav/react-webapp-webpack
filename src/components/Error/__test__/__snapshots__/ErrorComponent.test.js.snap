// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component should render - hasError=false 1`] = `
<div>
  <div>
    test
  </div>
</div>
`;

exports[`Error Component should render - hasError=true 1`] = `
<div>
  <div>
    <h1>
      Technical Error
    </h1>
    <div>
      <p>
         
        There are some technical error
         
      </p>
    </div>
    <div>
      <button>
        Exit
      </button>
    </div>
  </div>
</div>
`;
